enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,542 @@
+with simple test for each CNI we want to declare as supported.
+
+Starting with `bridge` test shall: install Multus, create NetworkAttachmentDefinition and deploy
+a Pod that should have additional interface attached. After verifying that the interface is present,
+test will try to access Pod's application using the bridge interface on the host to make sure there
+is a connectivity.
+
+Adding Multus to existing cluster will be tested by adding this scenario to existing upgrade test
+on rpm-based RHEL.
+
+Tests for other networking plugins will be designed and implemented when plugins are planned for support.
+
+## Graduation Criteria
+
+Multus CNI for MicroShift is targeted to be GA next release.
+
+### Dev Preview -> Tech Preview
+
+N/A
+
+### Tech Preview -> GA
+
+N/A
+
+### Removing a deprecated feature
+
+N/A
+
+## Upgrade / Downgrade Strategy
+
+Because both Multus and MicroShift RPMs will be built from the same spec file, they will share the
+same version and it is expected that they are updated together following MicroShift upgrade rules
+depending on type of operating system (ostree-based or regular RPM).
+
+Considering only the manifests, we know that on each start MicroShift will apply manifests forcefully
+overwriting any differences. However, MicroShift does not have any uninstall capabilities.
+If manifests ever change, for example some ConfigMap is renamed, then these old parts will
+keep existing in the database. This could have undesirable consequences of having two Multus DaemonSets
+if the name or the namespace of original DaemonSet changes. To make the transition to thick Multus
+smooth, we should not deviate from already existing resource names present in OpenShift's Multus
+manifests. This problem is not Multus specific - it is how MicroShift works and it is not part of this
+enhancement addressing this shortcoming.
+
+Because of the way Multus works (checks Annotations when needed, executed when networking must be
+set up, does not keep its own database beside cache) we can consider it to be mostly stateless.
+When Multus CNI binary is executed by kubelet/CRI-O it will setup Pod's network according to the
+configs and Annotations (not a database with "expected states"), therefore updating Multus does not
+require migration strategies.
+
+If the schema of NetworkAttachmentDefinitions CRD changes, instead of deploying a mutating webhook
+(which would use resources idling but actually perform actions on first start), we can just suggest
+users to update their CRs.
+
+## Version Skew Strategy
+
+Building Multus and MicroShift RPMs from the same spec file means Multus should be updated together
+with MicroShift which means there should not be any version skew between MicroShift and Multus.
+This might change with introduction of multi-node deployments of MicroShift.
+
+## Operational Aspects of API Extensions
+
+If Multus (or any delegate CNIs it executes) fails, a new Pod will be stuck in "ContainerCreating" status
```

Comment:
Perhaps a good troubleshooting step?

Reply:
Well, that's just it: Pod doesn't start because of the error, user runs `oc describe $POD` and in the events there's a message from multus.